src/main/ruby/core/truffle/interop.rb

```diff
@@ -36,6 +36,17 @@ def self.ruby_object_keys(object)
         end
       end
     end
+    
+    def self.lookup_symbol(name)
+      # TODO CS 9-Aug-17 I can't see to check if the method is defined
+      if method = Object.method(name) rescue nil
+        method
+      elsif Truffle::SymbolOperations.is_constant?(name) && Object.const_defined?(name)
+        Object.const_get(name)
+      else
```

Comment:
I couldn't make your idea work, @eregeon. We still seem to not quite define top-level methods in the same way as MRI!